### DISCUSSION TITLE
ICM showing offline while connected to the internet

### DISCUSSION THREAD

**Discussion by paulz9090** [InternetConnectionMonitor/icm-public#112] [Q&A] [answered]
I have been using and relying on ICM for years. Recently, it started showing a dropped connections when the connection was good. I have deleated the extension and downloaded a new copy. Same result. It is showing offline as I am writing this. I feel that it must be my system as I don't see others complaining.

**Comment by pawtech97** [comment]
I have completed the same steps.  Its shows offline as I reply.  I do not believe it is your system, if so my system is somehow messed up too.

**Comment by sharkei2018** [comment]
I'm noticing the same starting today 13th August 2024.  As I am sending this answer, I have a red cross at the top of my Google browser Version 127.0.6533.100 (Official Build) (64-bit).

**Comment by dmishh** [comment]
@paulz9090 @pawtech97 @sharkei2018 folks, can you please tell me the name and version of your operating system and ICM version?

Also, can you send me a screenshot from the Realtime monitor (open it from the main ICM window)?

**Comment by paulz9090** [comment]
I can't be sure, but this may have started when I installed Windows 11 on
July 4, 2024.

Paul Zimmerman

Name and version of your operating system:

Edition Windows 11 Pro
Version 23H2
Installed on ‎7/‎4/‎2024
OS build 22631.4037
Experience Windows Feature Experience Pack 1000.22700.1027.0


 ICM version:

5.2.0

[image: image.png]

**Reply by dmishh** [reply]
@paulz9090 

I'm running the same version of Windows on my PC:
_Edition Windows 11 Home
Version 23H2
Installed on ‎10/‎21/‎2022
OS build 22631.4037
Experience Windows Feature Experience Pack 1000.22700.1027.0_

I didn't observe any issues with the detection as you describe.
It may be caused by another extension blocking the monitoring.

**Here are a few options to track down the issue:**
**(1)** You can download [Brave browser](https://brave.com/download/), install ICM and try there
**(2)** You can try to enable ICM in incognito mode (see the picture below) by right-clicking the icon and pressing "Manage extension", then open an incognito window and close other Chrome windows

If ICM works fine, most likely it's due to something inside Chrome blocking ICM (like another extension)

![image](https://github.com/user-attachments/assets/0e57f11b-cd7d-4831-aeee-7305ff1e4e73)

**Comment by dmishh** [comment]
@paulz9090 @pawtech97 @sharkei2018 folks, here is how you can help to track down the issue. 

**Please, right-click on the ICM icon and then click on "offscreen.html"**

![image](https://github.com/user-attachments/assets/e311ea6f-1ee8-4711-bf8b-4889e48db1fb)

<hr/>

**A console will open (make sure Console tab is active), take a screenshot of it, so I can see the errors. It looks like this:**

![image](https://github.com/user-attachments/assets/b4fb4270-e2b1-4259-bbb0-07040dc2be14)

**Comment by sharkei2018** [comment]
Hi
If I right mouse click the ICM icon > Manage extension I don't get the option to Inspect Views offscreen

![ICM 1](https://github.com/user-attachments/assets/634da615-8644-483d-a153-2f45eb3d6388)

However, I was running another extension - Malwarebytes Browser Guard version 3.0.7 and when I switch that off, ICM starts to work as it did previously.  I'm not aware that MBG updated itself earlier this week, but it may have done.

For completeness my system is:
Dell Inspiron 3910
Windows 11 Home
Version 23H2
Installed on 29/11/2022
OS Build 22631.4037
Windows Feature Experience Pack 1000.22700.1027.0

Google Chrome Version 127.0.6533.120 (Official Build) (64-bit)

Looks like stopping using Malwarebytes resolves the ICM issue.  By the way, I have found that Malwarebytes can sometimes work in a strange way interacting with other software, so it's not a total surprise.

**Reply by dmishh** [reply]
@sharkei2018 Great, thank you very much for tracking it down!
This is what I expected.
**Another extension is blocking the monitoring**, in your case, it's Malwarebytes Browser Guard.

> I'm not aware that MBG updated itself earlier this week, but it may have done.

The extensions update silently on Chrome restart, so you don't notice anything.

> If I right mouse click the ICM icon > Manage extension I don't get the option to Inspect Views offscreen

Yeah, sorry, forgot to mention that you need to enable "Developer mode". It's on the Extensions page (chrome://extensions/) in the top right corner.

**Comment by dmishh** [comment, marked as answer]
Here is a way to solve this - #113
